### ISSUE TITLE
Class A ABP mode not receiving any messages

### ISSUE THREAD

**Issue by reyes73a** [arduino-libraries/MKRWAN#108]
Hi,

I have been trying to figure out why my setup is not currently receiving messages. I feel like I am doing something dumb somewhere. The end device uses the MKR1310 to communicate to a MultiTech Gateway in a Class A configuration. Uplink messages reach the gateway fine, but I cannot receive any downlink messages. I have the gateway automatically reply with a message whenever an uplink is received. However, on the end node side the method **_modem.available()_** is always **false** right after sending a message no matter what... I tried waiting a second, not waiting, etc. all with the same result. It makes no sense to me. For comparison purposes I have set up a lopy4 from pycom with the same configuration and equivalent code (send/receive) and the lopy can receive the downlink messages. Therefore, I am sure this is an MKR issue. I am sending the downlink messages to port 1 and on the MKR side I specify the same port using **_modem.setPort(1)_**. I am also calling **_modem.configureClass(CLASS_A)_** to make sure it is in Class A. Here is how I configure it:

```cpp
Serial.println("---> Initializing LoRa Manager");

    if (!modem.begin(US915_HYBRID))
    {
        Serial.println("Failed to start module");

        while (1)
        {
        }
    };

    if (modem.setADR(false))
        Serial.println("ADR was configured successfully");

    if (modem.dataRate(3))
        Serial.println("DR was configured successfully");

    modem.setTimeout(5000);
    if (modem.setPort(1))
        Serial.println("Port was configured successfully");

    //Configure Class
    if (modem.configureClass(CLASS_A))
        Serial.println("Class was configured successfully");

    
    nwkSKey = "XXXX";
    appSKey = "XXXX";

    devAddr.trim();
    nwkSKey.trim();
    appSKey.trim();

    int connected = modem.joinABP(devAddr, nwkSKey, appSKey);

    if (connected)
    {
        Serial.println("Connected to ABP succesfully");
    }
    else
    {
        Serial.println("Something went wrong; are you indoor? Move near a window and retry");

        while (1)
        {
        }
    }

    Serial.print("Your module version is: ");
    Serial.println(modem.version());
    Serial.print("Your device EUI is: ");
    Serial.println(modem.deviceEUI());
    Serial.print("Your device Addr is: ");
    Serial.println(modem.getDevAddr());
```
right after I send and try to receive a message like so:

```cpp
   modem.beginPacket();
    modem.write((short)0);
    modem.write((float)batteryVoltage);
    modem.write((float)runTimeInHrs);
    int err = modem.endPacket(true);

    if (err > 0)
    {
        Serial.println("Message sent correctly!");
    }
    else
    {

        Serial.println("Error sending message :(");
    }

    delay(1000);

    char rcv[256];
    int i = 0;

    while (modem.available())
    {
        Serial.println("Modem is available");
        rcv[i++] = (char)modem.read();
    }

    Serial.print("Received: ");
    Serial.print(i);
    Serial.print(" bytes");
    Serial.println("");
```

 Any thoughts on what I might be missing? 

Thanks!!

**Comment by reyes73a** [comment]
nobody has any thoughts on this??

**Comment by reyes73a** [comment]
FYI: looking a bit more into this. By looking at the gateway log I can see the ABP uplink message arriving and then the downlink message leaving the gateway. In fact, the gateway removes the downlink message from the queue, but the MKR always returns 0 for modem.available() right after sending an uplink suggesting that there is nothing in the rx buffer which makes no sense...

**Comment by reyes73a** [comment]
Update: finally got ABP message reception working on two different MKR boards. It seems (although I am not 100%) that the cause of this was either related to the gateway configuration or the fact that I updated the modem and arduino firmware to match at 1.2.4. During testing I did mix and match device addresses with DEVUIs while testing ABP and OTAA at the same time. It could have been that the gateway configuration was messed up because of this. After deleting the device profiles and creating new ones it seems to receive downlinks for both OTAA and ABP. I still have to test and confirm on the master branch, but this version: [https://github.com/arduino-libraries/MKRWAN/actions/runs/1398096763/workflow](url) seems to work.
